### PR TITLE
Earfile bucket

### PR DESCRIPTION
**Pull request by tony-griffin**
There are now two buckets for Was, setup-scripts and artifacts

One contains the bootstrap scripts and the other the EAR file for the B7 application.
When Was boots, it pulls in both the setup scripts and the artifacts and provisions the box and installs the application.

A future PR will refactor the bootstrapping to use the new module.

### CODE REVIEW

File: modules/ec2/main.tf [hunk 1]
```diff
@@ -17,31 +15,61 @@ resource "aws_ssm_parameter" "instance_private_key" {
   type        = "SecureString"
   value       = tls_private_key.ec2.private_key_pem
   description = "master ssh key for env ${var.name_prefix}"
+  overwrite   = true
```
Comment (emileswarts) [inline modules/ec2/main.tf:18]: Ran into Terraform issues, this flag is required to overwrite the private key which would be an error otherwise.

Comment (cob16) [reply modules/ec2/main.tf:17]: Why are we overwriting this in the first place? Does this need to be a perm fix?

File: modules/ec2/main.tf [hunk 2]
```diff
@@ -16,31 +14,61 @@ resource "aws_ssm_parameter" "instance_private_key" {
...
+resource "aws_s3_bucket" "setup-scripts" {
+  bucket = "${var.name_prefix}-setup-scripts"
   acl    = "private"
 
   force_destroy = true
 
   versioning {
     enabled = true
   }
+// Disabled until we have the AWS SDK installed and configured
```
Comment (cob16) [inline modules/ec2/main.tf:31]: This is working using only curl for me and @CaitBarnard. The one gotcha is you must reupload the config after this change is applied as this is only the default encryption

File: modules/ec2/main.tf [hunk 3]
```diff
@@ -16,31 +14,61 @@ resource "aws_ssm_parameter" "instance_private_key" {
...
+      Version = "2012-10-17"
+    }
+  )
+}
+
+resource "aws_kms_key" "setup-scripts-key" {
+  description = "${var.name_prefix}-S3-setup-scripts-key"
+  tags        = var.tags
+}
+
```
Comment (cob16) [inline modules/ec2/main.tf:71]: these resources have been added by mistake in the merge

File: modules/ec2/main.tf [hunk 4]
```diff
@@ -16,31 +14,61 @@ resource "aws_ssm_parameter" "instance_private_key" {
...
+      ]
+      Version = "2012-10-17"
+    }
+  )
+}
+
+resource "aws_kms_key" "setup-scripts-key" {
+  description = "${var.name_prefix}-S3-setup-scripts-key"
+  tags        = var.tags
+}
```
Comment (cob16) [inline modules/ec2/main.tf:70]: these lines are not needed this policy is defined below in the policy docs

File: modules/ec2/main.tf [hunk 5]
```diff
@@ -113,10 +144,39 @@ resource "aws_s3_bucket_policy" "installer_bucket" {
...
+
+//  server_side_encryption_configuration {
+//    rule {
+//      apply_server_side_encryption_by_default {
+//        kms_master_key_id = aws_kms_key.artifact-store-key.arn
+//        sse_algorithm     = "aws:kms"
+//      }
+//    }
+//  }
+
```
Comment (cob16) [inline modules/ec2/main.tf:171]: see comment above

File: modules/ec2/main.tf [hunk 6]
```diff
@@ -113,10 +144,39 @@ resource "aws_s3_bucket_policy" "installer_bucket" {
...
+  tags = var.tags
+}
+
+resource "aws_kms_key" "artifact-store-key" {
+  description = "${var.name_prefix}-S3-artifact-store-key"
+  tags        = var.tags
+}
```
Comment (cob16) [inline modules/ec2/main.tf:178]: This resource is no longer used

File: modules/was/main.tf [hunk 7]
```diff
@@ -75,6 +75,19 @@ resource "aws_security_group" "was" {
   tags = local.was_tags
 }
 
+resource "random_password" "was" {
+  length  = 24
+  special = false
+}
+
+data "template_file" "install_was" {
+  template = "${file("${path.module}/install_was.sh.tpl")}"
```
Comment (cob16) [inline modules/was/main.tf:84]:
```suggestion
  template = file("${path.module}/install_was.sh.tpl")
```

File: modules/ec2/main.tf (hunk 1 above)
Comment (cob16) [reply modules/ec2/main.tf:17]: after trying to remove it this problem persists it seems

**Pull request merged**